### ISSUE TITLE
Full GOROOT paths in Go 1.23 binaries

### ISSUE THREAD

**Issue by JacobOaks** [bazelbuild/rules_go#3983]
### What version of rules_go are you using?

v0.48.0

### What version of gazelle are you using?

v0.37.0

### What version of Bazel are you using?

7.2.0

### Does this issue reproduce with the latest releases of all the above?

Yes

### What operating system and processor architecture are you using?

linux amd64

### Any other potentially useful information about your toolchain?

N/A

### What did you do?

Ran this rules_go test on Go 1.23rc1:
```shell
$ bazel test //tests/core/strip:strip_test
```

### What did you expect to see?

Target passes.

### What did you see instead?

All cases failed with messages like the following. For brevity I'll only show one case.
```
    --- FAIL: Test/run_sometimes_opt (0.95s)
        strip_test.go:228: running: bazel run --strip sometimes --compilation_mode opt //:strip -- -wantstrip=false
        strip_test.go:240: unexpected error: exit status 2
            stderr:
            WARNING: The following rc files are no longer being read, please transfer their contents or import their path into one of the standard rc files:
            /etc/bazel.bazelrc
            Computing main repo mapping:
            Loading:
            Loading: 0 packages loaded
            WARNING: Build option --compilation_mode has changed, discarding analysis cache (this can be expensive, see https://bazel.build/advanced/performance/iteration-speed).
            Analyzing: target //:strip (0 packages loaded, 0 targets configured)
            Analyzing: target //:strip (0 packages loaded, 0 targets configured)
            [0 / 1] [Prepa] BazelWorkspaceStatusAction stable-status.txt
            INFO: Analyzed target //:strip (0 packages loaded, 10813 targets configured).
            INFO: Found 1 target...
            Target //:strip up-to-date:
              bazel-bin/strip_/strip
            INFO: Elapsed time: 0.908s, Critical Path: 0.26s
            INFO: 2 processes: 1 internal, 1 processwrapper-sandbox.
            INFO: Build completed successfully, 2 total actions
            INFO: Running command line: bazel-bin/strip_/strip '-wantstrip=false'
            panic: got unexpected stack trace line "\tbazel-out/k8-opt/bin/external/io_bazel_rules_go/stdlib_/src/runtime/debug/stack.go:26 +0x5e" at index 2

            goroutine 1 [running]:
            main.main()
                strip.go:33 +0x2b9
```

This is likely due to https://github.com/golang/go/issues/62047, which drops support for `GOROOT_FINAL`, which rules_go uses to replace actual GOROOTs to keep builds reproducible (see #971).
